Add RTD Documentation

Bachibouzouk opened the pull request:
To build the documentation locally:
1.  create/activate your virtual environment
2. move to the `docs` folder with `cd docs`
3. install the doc dependencies (only have to do it once): `pip install -r docs-requirements.txt`
4. build the documentation locally: `make html` (you can then in your file explorer open the file `docs/build/html/index.html` which should automatically open in your favorite browser)

Bachibouzouk commented on README.rst:
```diff
@@ -0,0 +1,184 @@
+
+.. code-block:: python
+
+  pip install ramp
```
`pip install ramp` will not install this package but

https://pypi.org/project/ramp/

Bachibouzouk replied:
If you want to install the local package one can `pip install -e .` when being at the root of the repository

mohammadamint replied:
> `pip install ramp` will not install this package but
> 
> https://pypi.org/project/ramp/

Sure. Just put at as a template to be filled correctly once we have ramp uploaded on pypi.

Bachibouzouk commented on README.rst:
```diff
@@ -0,0 +1,184 @@
+.. code-block:: python
+
+   # adding some appliances for the household
+   radio = low_income_household.Appliance(
```
```suggestion
   radio = low_income_household.add_appliance(
```
`.Appliance` should only be used for back-compatibility and `.add_appliance` should be preferred

Bachibouzouk commented:
I suggest we focus on commenting what is not working in the proposed version in this PR and that we postpone changes of structure or addition of explanation in subsequent PR, otherwise we might end up spending a lot of time merging this PR. I know a way to automatically generate the readthedocs online upon pushing commits, I could set that up (I will see if I have enough rights for that)

FLomb commented:
> To build the documentation locally:
> 
> 1. create/activate your virtual environment
> 2. move to the `docs` folder with `cd docs`
> 3. install the doc dependencies (only have to do it once): `pip install -r docs-requirements.txt`
> 4. build the documentation locally: `make html` (you can then in your file explorer open the file `docs/build/html/index.html` which should automatically open in your favorite browser)

This works for me, but it is worth mentioning that I end up with two warnings in the process. See below:
```
/home/fl/GitHub-repos/RAMP/ramp/core/core.py:docstring of ramp.core.core.Appliance.specific_cycle:12: WARNING: Inline strong start-string without end-string.
/home/fl/GitHub-repos/RAMP/docs/source/examples/using_excel/using_excel.rst:229: WARNING: image file not readable: examples/using_excel/excel_file.png
```
I guess this affects what I see in a couple of pages of the docs. Other than this, I have a few comments about the contents and the structure which I will provide where/when needed

FLomb commented on README.rst:
```diff
@@ -0,0 +1,184 @@
+
+   conda create -n ramp python=3.8
+
+2. If you create a new environment for mario, to use it, you need to activate the ramp environment each time by writing
```
there is a reference to 'mario' here which I guess is a typo

FLomb commented on README.rst:
```diff
@@ -0,0 +1,184 @@
+conda package manager which can install the current and future RAMP
+depencies in an easy and user friendly way.
+
+To get conda, `download and install "Anaconda Distribution" <https://www.anaconda.com/products/individual>`_.
```
Perhaps we should also point to miniconda, which is lighter

FLomb commented on README.rst:
```diff
@@ -0,0 +1,184 @@
+==========================
+
+
+Installing throguh source code
```
How?

mohammadamint replied:
Will be added.

FLomb commented on README.rst:
```diff
@@ -0,0 +1,184 @@
+
+  pip install ramp
+
+Installing through conda
```
Why missing? we already had text about how to install via conda in the development branch

mohammadamint replied:
more than installing through the txt file, we may consider publish it on conda-forge

FLomb commented on README.rst:
```diff
@@ -0,0 +1,184 @@
+
+Requirements
+------------
+RAMP has been tested on macOS and Windows.
```
And Linux too

FLomb commented on README.rst:
```diff
@@ -0,0 +1,184 @@
+For running RAMP, a couple of things are needed:
+
+#. The Python programming language, version 3.6 or higher
+#. A number of Python adds-on packages
```
Why don't we have the specific packages' names anymore? If we don't want to list them here, we should at least point to where they can be found

mohammadamint replied:
We may merge this part and python requirements into one section.

FLomb commented on README.rst:
```diff
@@ -0,0 +1,184 @@
+
+#. The Python programming language, version 3.6 or higher
+#. A number of Python adds-on packages
+#. RAMP software itself
```
I would say this is redundant, isn't it?

FLomb commented on README.rst:
```diff
@@ -0,0 +1,184 @@
+Quick start
+------------
+
+How to create a user category?
```
I don't think it's ideal to begin the 'Quick start' with user categories. What does even 'user category' mean to someone who lends on this page for the first time? 

We need to start from the broader context, as we do in the development branch. Why is all that text gone? Starting from a broader and simple walkthrough, we can then point users to more detailed examples to be found in the docs.

mohammadamint replied:
> I don't think it's ideal to begin the 'Quick start' with user categories. What does even 'user category' mean to someone who lends on this page for the first time?
> 
> We need to start from the broader context, as we do in the development branch. Why is all that text gone? Starting from a broader and simple walkthrough, we can then point users to more detailed examples to be found in the docs.

Makes sense. I will merge this new quick start with the previous version of README to provide a better understanding of the context.

FLomb commented on README.rst:
```diff
@@ -0,0 +1,184 @@
+
+Python module requirements
+--------------------------
+Some of the key packages that RAMP relies on are:
```
Ok, we already mention some vague (see my comment above) requirements in lines 72-73. Why do we repeat the requirements here, but now in a more expanded fashion? And why are they still less than those listed in the development branch?

mohammadamint replied:
I avoided the repetition in the new series of commits.

Regarding having less dependencies listed here with respect to development branch, as math and random are built in python packages, I avoided mentioning them explicitly as dependencies.

FLomb commented on README.rst:
```diff
@@ -0,0 +1,184 @@
+*F. Lombardi, S. Balderrama, S. Quoilin, E. Colombo, Generating high-resolution multi-energy load profiles for remote areas with an open-source stochastic model, Energy, 2019,*
+`https://doi.org/10.1016/j.energy.2019.04.097 <https://doi.org/10.1016/j.energy.2019.04.097>`_
+
+List of publications
```
This is now on the website, so we can get rid of it and instead point users to the website.
In general, we should make sure that both the website and the Gitter chat are shown in a prominent position at the top of this README file

mohammadamint replied:
> 

Good idea. I will make sure that all the badges (gitter, pypi, conda, docs, website,...) will be position at the top of the readme file.

FLomb commented on ramp/core/constants.py:
```diff
@@ -108,7 +108,7 @@ def switch_on_parameters():
             "number",
             "power",
-            "p_series",
+            #"p_series",
```
what's going on here?

FLomb reviewed (changes requested):
Thanks for kick-starting these docs, it's great to finally have something professional!

There are a number of content-related issues to fix before merging. In general, the landing page should contain less information and redirect more to the other pages where more details can be found.

The new example notebooks are very useful, but they need to be put into context. How do they relate to the example input files? (which are, incidentally, missing from the docs, whilst they should be there)

Bachibouzouk commented on docs/source/examples.rst:
```diff
@@ -0,0 +1,12 @@
+RAMP examples
+======================================
```
```suggestion
RAMP examples
=============
```
The line under the title should have the same character length as the text

Bachibouzouk commented:
We might want to double check the structure of the docs (section, subsection etc) to make sure we use the syntax in the correct steps
https://raw.githubusercontent.com/readthedocs/sphinx_rtd_theme/f854c890e1445b99a43ac5f10463a5df83739a2e/docs/demo/structure.rst

Another project I worked on, where we spent quite some time on the structure could help for inspiration
https://github.com/rl-institut/multi-vector-simulator/tree/dev/docs for the source and https://multi-vector-simulator.readthedocs.io/en/latest/index.html for the result

Bachibouzouk commented:
The RAMP website url should be found on the landing docs page

Bachibouzouk commented:
https://pyvideo.org/write-the-docs-2016/write-the-readable-readme.html could be a nice inspiration :)

Bachibouzouk commented:
By using `entry_points` in `setup.py` one can then run ramp from the command line from anywhere 

```
entry_points={  # Optional
    "console_scripts": [
        "ramp=ramp.cli:main",
    ],
},
``` 

I only moved the file `ramp.py` into the `ramp/` folder, renamed it `cli.py` and moved the code which was under `if __name__ == "__main__":` to a function called `main()`

Then after I reinstall ramp locally `pip install -e .` I can then run very simply `ramp -i <path to input file> ` instead of `python ramp.py -i <path to input file>` (which I could only run from the root of this repository anyway)

Bachibouzouk commented:
https://github.com/rl-institut/super-repo/issues/8 could also be used for inspiration of readme file

mohammadamint commented:
> By using `entry_points` in `setup.py` one can then run ramp from the command line from anywhere
> 
> ```
> entry_points={  # Optional
>     "console_scripts": [
>         "ramp=ramp.cli:main",
>     ],
> },
> ```
> 
> I only moved the file `ramp.py` into the `ramp/` folder, renamed it `cli.py` and moved the code which was under `if __name__ == "__main__":` to a function called `main()`
> 
> Then after I reinstall ramp locally `pip install -e .` I can then run very simply `ramp -i <path to input file> ` instead of `python ramp.py -i <path to input file>` (which I could only run from the root of this repository anyway)
> 
> I can push this as commits on this PR or add it once it is merged so we avoid potential conflicts

Thanks for making my work easier!

Bachibouzouk commented on Pipfile:
```diff
@@ -0,0 +1,13 @@
+name = "pypi"
+
+[packages]
+ramp = {editable = true, path = "."}
```
Isn't it rampdemand here as well?

Bachibouzouk commented:
@mohammadamint - when do you think you will merge this pull request?
